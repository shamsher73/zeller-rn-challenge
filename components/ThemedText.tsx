import { Text, type TextProps, StyleSheet } from 'react-native';

export type ThemedTextProps = TextProps & {
  lightColor?: string;
  darkColor?: string;
  type?: 'default' | 'title' | 'primary' | 'secondary';
};

export function ThemedText({
  style,
  lightColor,
  darkColor,
  type = 'default',
  ...rest
}: ThemedTextProps) {

  return (
    <Text
      style={[
        type === 'default' ? styles.default : undefined,
        type === 'title' ? styles.title : undefined,
        type === 'primary' ? styles.primary : undefined,
        type === 'secondary' ? styles.secondary : undefined,
        style,
      ]}
      {...rest}
    />
  );
}

const styles = StyleSheet.create({
  default: {
    fontSize: 16,
    fontWeight: '400',
  },
  primary: {
    fontSize: 16,
    fontWeight: '400',
  },
  title: {
    fontSize: 22,
    fontWeight: '500',
  },
  secondary: {
    fontSize: 12,
    fontWeight: '300',
  },
  link: {
    lineHeight: 30,
    fontSize: 16,
    color: '#0a7ea4',
  },
});
