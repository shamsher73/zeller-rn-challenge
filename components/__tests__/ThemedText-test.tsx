
import { render } from '@testing-library/react-native';
import { ThemedText } from '../ThemedText';

describe('<ThemedText />', () => {
    test('Text renders correctly on ThemedText', () => {
        const { getByText } = render(<ThemedText>Rabbit</ThemedText>);

        getByText('Rabbit');
    });

    test('snapshot', () => {
        const tree = render(<ThemedText>Rabbit</ThemedText>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
