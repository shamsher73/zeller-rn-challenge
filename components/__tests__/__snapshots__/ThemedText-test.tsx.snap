// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemedText /> snapshot 1`] = `
<Text
  style={
    [
      {
        "fontSize": 16,
        "fontWeight": "400",
      },
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Rabbit
</Text>
`;
